app/scripts/lib/multichain-api/provider-authorize/handler.js

```diff
@@ -73,25 +119,44 @@ export async function providerAuthorizeHandler(req, res, _next, end, hooks) {
       },
     );
     const permittedAccounts = getAccountsFromPermission(subjectPermission);
-    const { flattenedRequiredScopes, flattenedOptionalScopes } = processScopes(
-      requiredScopes,
-      optionalScopes,
-      { findNetworkClientIdByChainId },
+    assignAccountsToScopes(supportedRequiredScopes, permittedAccounts);
+    assignAccountsToScopes(supportedOptionalScopes, permittedAccounts);
+
+    const sessionScopes = mergeScopes(
+      supportedRequiredScopes,
+      supportedOptionalScopes,
     );
 
-    Object.keys(flattenedRequiredScopes).forEach((scope) => {
-      if (scope !== 'wallet') {
-        flattenedRequiredScopes[scope].accounts = permittedAccounts.map(
-          (account) => `${scope}:${account}`,
-        );
-      }
-    });
-    Object.keys(flattenedOptionalScopes).forEach((scope) => {
-      if (scope !== 'wallet') {
-        flattenedOptionalScopes[scope].accounts = permittedAccounts.map(
-          (account) => `${scope}:${account}`,
-        );
-      }
+    await Promise.all(
+      Object.keys(scopedProperties || {}).map(async (scopeString) => {
+        const scope = sessionScopes[scopeString];
+        if (!scope) {
+          return;
+        }
+
+        const networkClientId = await validateAndUpsertEip3085({
+          scopeString,
+          eip3085Params: scopedProperties[scopeString].eip3085,
+          origin,
+          upsertNetworkConfiguration: hooks.upsertNetworkConfiguration,
+          findNetworkClientIdByChainId: hooks.findNetworkClientIdByChainId,
+        });
+
+        if (networkClientId) {
+          networkClientIdsAdded.push(networkClientId);
+        }
+      }),
+    );
+
+    assertScopesSupported(sessionScopes, {
```

Comment:
and we're doing another round of validating that there is a networkClientId? It seems like we're doing this 3 times in this handler? I guess I can see we want to validate the upserts worked? 

I was thinking it could make sense to go back and remove the failed upserts from `sessionScopes`? That would seem a more graceful way of handling this case than throwing if an upsert fails

Reply:
I suppose this assert is not necessary actually. Upserts failing would cause the entire request to fail

Reply:
removed https://github.com/MetaMask/metamask-extension/pull/25873/commits/8179eb93c6d0e39066084e32a9b1c672c246fc26